core/src/main/java/com/flipkart/poseidon/utils/ApiHelper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2015 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.poseidon.utils;
+
+import org.springframework.http.HttpMethod;
+
+/**
+ * Created by venkata.lakshmi on 04/02/16.
+ *
+ *  Helper class which contains all util functions
+ */
+public class ApiHelper {
+
+    public static String getUrlWithHttpMethod(String url, HttpMethod httpMethod) {
+        if (httpMethod == null) {
+            return url;
+        }
+        return String.format("%s/_%s_", url, httpMethod);
```

Comment:
What if the url already ended with /. Trie behaviour might change for /user/__GET__ vs /user//__GET__